### PR TITLE
Tower of Hanoi Algorithm - C

### PR DESCRIPTION
**Pull request by SangeetaMishra**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

(Write your answer here.)

### Description

(Write your answer here.)

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#793

### CODE REVIEW

File: C/other/Tower_Of_Hanoi.c [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        printf("%d -> %d\n", a,b);
+        return ;
+    }
+    TowerOfHanoi(n-1,a,c,b);
+    printf("%d -> %d\n", a,b);
+    TowerOfHanoi(n-1,c,b,a);
+}
+
+int main(){
+    int n;
```
Comment (siddharth25pandey) [inline C/other/Tower_Of_Hanoi.c]: Better to display message what is n for user

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]

File: C/other/Tower_Of_Hanoi.c (hunk 1 above)
Comment (siddharth25pandey) [reply C/other/Tower_Of_Hanoi.c]: Please display message before & after of n.

**Review by siddharth25pandey — CHANGES_REQUESTED** [review]

File: C/other/Tower_Of_Hanoi.c [hunk 2]
```diff
@@ -0,0 +1,42 @@
+#include <stdio.h>
+
+void TowerOfHanoi(int n,int a,int b,int c){
```
Comment (iamrajiv) [inline C/other/Tower_Of_Hanoi.c:3]: @SangeetaMishra this code can be shorten using this
```
void TowerOfHanoi(int n, int a, int b, int c)
{
    if (n != 0)
    {
        TowerOfHanoi(n - 1, a, c, b);
        printf("%d -> %d\n", a, b);
        TowerOfHanoi(n - 1, c, b, a);
    }
}
```

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@SangeetaMishra Add space and time complexity. Also, add a little bit description of the Tower of Hanoi problem and mention that the recursive method is used in this program.

File: C/other/Tower_Of_Hanoi.c [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+        printf("%d -> %d\n", a,b);
+        return ;
+    }
+    TowerOfHanoi(n-1,a,c,b);
+    printf("%d -> %d\n", a,b);
+    TowerOfHanoi(n-1,c,b,a);
+}
+
+int main(){
+    int n;
```
Comment (Abhijit2505) [inline C/other/Tower_Of_Hanoi.c]: Change the variable name n to something good, like number_of_discs or something like that.
It's necessary for a code to speak itself.

**Review by Abhijit2505 — CHANGES_REQUESTED** [review]
Please make the required changes.

**Review by siddharth25pandey — APPROVED** [review]
@SangeetaMishra LGTM
Able to merge
Thanks!

**Review by iamrajiv — APPROVED** [review]

**Comment by SangeetaMishra** [comment]
@siddharth25pandey  @iamrajiv  @Abhijit2505  Could you please approve the pull request as I have to make other pull requests?

**Review by iamrajiv — APPROVED** [review]

**Review by atarax665 — APPROVED** [review]

**Pull request merged**